### ISSUE TITLE
Require valid rule categories

### ISSUE THREAD

**Issue by not-an-aardvark** [not-an-aardvark/eslint-plugin-eslint-plugin#60]
This rule would require that all lint rules have a `meta.docs.category` property which is one of several user-specified strings. The goal would be to prevent mistakes like the one pointed out in https://github.com/eslint/eslint/pull/10661#discussion_r204618870.

Invalid example:

```js
/* eslint eslint-plugin/require-valid-categories: [error, ["Possible Errors", "Stylistic Issues"]] */

module.exports = {
    meta: {
        docs: {
            category: "Not a valid category"
        }
    },
    create(context) {
        // ...
    }
};
```

Valid example:

```js
/* eslint eslint-plugin/require-valid-categories: [error, ["Possible Errors", "Stylistic Issues"]] */

module.exports = {
    meta: {
        docs: {
            category: "Possible Errors"
        }
    },
    create(context) {
        // ...
    }
};
```
